### PR TITLE
Wave test stability check

### PR DESCRIPTION
**Pull request by majosm**
Adds stability check to wave tests as requested in #15.

### CODE REVIEW

**Comment by majosm** [comment]
Confirmed that tests detect instability. Now adding upwind flux fix...

File: test/test_wave.py [hunk 1]
```diff
@@ -78,89 +79,149 @@ def mesh_factory(n):
...
+    u = sym_eval(sym_u, t)
+    v = sym_eval(sym_v, t)
+
+    fields = flat_obj_array(u, v)
+
+    from mirgecom.integrators import rk4_step
+    dt = timestep_scale/order**2
+    for istep in range(10):
+        fields = rk4_step(fields, t, dt, get_rhs)
+        t += dt
```
Comment (inducer) [inline test/test_wave.py:207]: Did this fail before the fix? (Just making sure.)

File: test/test_wave.py [hunk 2]
```diff
@@ -78,89 +79,149 @@ def mesh_factory(n):
...
+                            ("rhs_v_actual", rhs[1:]),
+                            ("rhs_u_expected", expected_rhs[0]),
+                            ("rhs_v_expected", expected_rhs[1:]),
                         ])
 
     print("Approximation error:")
     print(eoc_rec)
     assert(eoc_rec.order_estimate() >= order - 0.5 or eoc_rec.max_error() < 1e-11)
+
+    # Check stability
```
Comment (inducer) [inline test/test_wave.py:179]: Could you make stability a separate test?

File: test/test_wave.py (hunk 1 above)
Comment (majosm) [reply test/test_wave.py:207]: Yep! See [here](https://github.com/illinois-ceesd/mirgecom/runs/851228389), e.g.:

```
Traceback (most recent call last):
187
  File "/home/runner/work/mirgecom/mirgecom/test/test_wave.py", line 227, in test_wave
188
    assert(err < max_err)
189
AssertionError: assert 116748.05780039357 < 1.8771041427775335
```

File: test/test_wave.py (hunk 2 above)
Comment (majosm) [reply test/test_wave.py]: Done.

File: test/test_wave.py [hunk 3]
```diff
@@ -81,29 +91,62 @@ def mesh_factory(n):
...
+    # rhs(v part) = c*grad(u)
+    sym_rhs = flat_obj_array(
+                sym_c * sym.div(sym_v) + sym_f,
+                make_obj_array([sym_c]) * sym.grad(dim, sym_u))
+
+    return sym_u, sym_v, sym_f, sym_rhs
+
+
+def max_inf_norm(fields):
+    return np.max(np.array([la.norm(field.get(), np.inf) for field in fields]))
```
Comment (inducer) [inline test/test_wave.py:125]: FYI: On the array context branch, there's a `discr.norm`.

**Pull request merged**